tests/e2e/fundAccountPage.test.ts

```diff
@@ -70,163 +81,271 @@ test.describe("[futurenet/testnet] Fund Account Page", () => {
     await expect(page.getByText("Fund", { exact: true })).toBeDisabled();
   });
 
-  test.skip("Successfully funds an account when clicking 'Get lumens' with a valid public key", async ({
-    page,
-  }) => {
-    const publicKeyInput = page.locator("#fund-public-key-input");
-    const getLumenButton = page
-      .getByTestId("fundAccount-buttons")
-      .getByText("Get lumens");
+  test("Fund new account with XLM and add USDC trustline", async ({ page }) => {
+    let accountsCallCount = 0;
 
-    await publicKeyInput.fill(
-      "GB6D5E2HRVBD6QIOXCE6ILMUFQT45LMSZSKRZUCL4ZK6Q6GCNMTJ2E3C",
+    // Mock accounts API call
+    await page.route(
+      `*/**/accounts/${TEST_ACCOUNT_PUBLIC_KEY}`,
+      async (route) => {
+        if (accountsCallCount === 0) {
+          await route.fulfill({
+            status: 200,
+            contentType: "application/hal+json; charset=utf-8",
+            body: JSON.stringify(MOCK_ACCOUNT_UNFUNDED_RESPONSE),
+          });
+        } else if (accountsCallCount === 1) {
+          await route.fulfill({
+            status: 200,
+            contentType: "application/hal+json; charset=utf-8",
+            body: JSON.stringify(MOCK_ACCOUNT_FUNDED_RESPONSE),
+          });
+        } else {
+          await route.fulfill({
+            status: 200,
+            contentType: "application/hal+json; charset=utf-8",
+            body: JSON.stringify(MOCK_ACCOUNT_FUNDED_RESPONSE_WITH_USDC),
+          });
+        }
+
+        accountsCallCount++;
```

Comment:
I don't understand this part `accountsCallCount++`. Does test get re-run after `accountsCallCount++` is called?